### PR TITLE
re-Sidekiq Workers

### PR DESCRIPTION
**Pull request by andela-oakinniranye**
SideKiq gets to run before some resources are fully persisted to the database, which often results in a record not found error.
Changed the perform_async method to instead perform in 5 seconds(which is a reasonable time for a record to have been persisted).
Also added the rspec-sidekiq gem so as to enable testing with the sidekiq helpers/matchers.

### CODE REVIEW

File: config/newrelic.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 common: &default_settings
   # Required license key associated with your New Relic account.
-  license_key: 72d898b4665ce655ede38eedf0aea02192d5f3e3
+  license_key: <%= ENV["NEW_RELIC_LICENSE_KEY"] %>
```
Comment (andela-oojewale) [inline config/newrelic.yml]: Lol

**Comment by andela-oojewale** [comment]
👍

File: app/models/concerns/new_notification.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module NewNotification
   end
 
   def notify
-    NotificationSystemWorker.perform_async(model_name.to_s, id)
+    NotificationSystemWorker.perform_in(5.seconds, model_name.to_s, id)
```
Comment (andela-ooranagwa) [inline app/models/concerns/new_notification.rb]: Learning time: What's the advantage of `perform_in`?

File: config/newrelic.yml (hunk 1 above)
Comment (andela-ooranagwa) [inline config/newrelic.yml]: 🙈

**Comment by andela-ooranagwa** [comment]
👍

**Pull request merged**